### PR TITLE
Add storage role assignment to extension tests

### PR DESCRIPTION
**Pull request by pakrym**
To try and fix weekly failures.

### CODE REVIEW

File: sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs [hunk 1]
```diff
@@ -60,16 +60,26 @@ protected TestEnvironment()
...
             _prefix = serviceName.ToUpperInvariant() + "_";
 
-            var testEnvironmentFile = Path.Combine(serviceSdkDirectory, "test-resources.json.env");
-            if (File.Exists(testEnvironmentFile))
+            var testEnvironmentFiles = new[]
             {
-                var json = JsonDocument.Parse(
-                    ProtectedData.Unprotect(File.ReadAllBytes(testEnvironmentFile), null, DataProtectionScope.CurrentUser)
-                );
+                Path.Combine(serviceSdkDirectory, "test-resources.bicep.env"),
```
Comment (pakrym) [inline sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs:65]: reacts to https://github.com/Azure/azure-sdk-tools/pull/1876

**Review by JoshLove-msft — APPROVED** [review]

**Pull request merged**